js/build.js

```diff
@@ -2690,10 +2690,16 @@ Fliplet.Widget.instance('chat', function (data) {
         return email;
       }
 
+      var options = {
+        crossLoginColumnName: crossLoginColumnName
+      };
+
       // Fallback to use session data as the usage of
       // the fl-chat-auth-email app storage is deprecated.
-      return Fliplet.User.getCachedSession().then(function (session) {
-        email = _.get(session, ['entries', 'dataSource', 'data', crossLoginColumnName]);
+      return Fliplet.Hooks.run('flChatGetUserEmail', options).then(function () {
```

Comment:
shouldn't this be called `beforeChatLogin` or similar?

Reply:
No, this is not for logging in, but for verifying that the user is logged in by getting the user's email.

Reply:
If you're concerned that it doesn't have `before` in the hook name, that can be added.

Reply:
@tonytlwu 

- `fl` is unnecessary, e.g. `beforeFormSubmit`
- use `before` or `after` to add more context
- `GetUserEmail` doesn't really tell me anything about when this hooks runs or what I am meant to do with it
- You are passing options so technically this is to configure the chat before it initialised. Do you agree? If so, name the hook accordingly as we have done on other widgets

Reply:
Please confirm if your concern is to do with the keyword `login` or that it doesn't have `before` as part of the hook name.

Reply:
@tonytlwu I've expressed all my concerns above. The hook name is unclear and I don't want this merged in its current state. Please name it so that it's clear at which point this runs and what it's meant to do.

Reply:
1. Leaving `fl` out changes the convention we have used throughout this component. This would create confusing naming across the documentation for this feature. Moreover, we also use `fl` in the LFD hooks.
1. I'll add `before`
1. This is not to configure the chat before it's initialised. It's just to change how the user email is retrieved (the only parameter being the column name that is changeable).

Reply:
@tonytlwu 
3. you are technically changing a parameter of the chat initial configuration like on `flListDataBeforeGetData`

Reply:
@squallstar Yes, but I'm not letting the user change anything else apart from the `crossLoginColumnName`, because the purpose is only for changing how the user email is retrieved.

If I had provided the entire configuration object for developers to customize, then I would've named it differently, and also run the hook in a different location.